web-frontend/src/pages/Home/Wallet/Wallet.tsx

```diff
@@ -0,0 +1,143 @@
+  FiUsers,
+  FiDisc,
+  FiSettings,
+  FiSun,
+} from 'react-icons/fi';
+import { useLocation, useNavigate } from 'react-router-dom';
+import useResource from '../../../custom/api/useResource';
+import { AccountObject } from '../../../models/AccountModel';
+import { routeFor } from '../../../utils';
+import { useState } from 'react';
+
+enum MenuItem {
+  Home = 'home',
+  SendReceive = 'send-receive',
+  Transactions = 'transactions',
+  Contacts = 'contacts',
+  Assets = 'assets',
+  Settings = 'settings',
+  LightTheme = 'light-theme',
+}
+
+interface WalletProps {
+  children: ReactNode;
+}
+
+function Wallet(props: WalletProps) {
+  const navigate = useNavigate();
+  const { state } = useLocation();
+  var nickname: string = state.nickname;
+
+  const { data: accounts = [] } = useResource<AccountObject[]>('accounts');
+
+  // If no account, redirect to welcome page
+  if (!accounts.length) {
+    navigate(routeFor('index'));
+  }
+
+  // const [selectedAccount, setSelectedAccount] = useState<AccountObject>(
+  //   accounts.find((account) => account.nickname === nickname)!,
+  // );
+
+  const [selectedItem, setSelectedItem] = useState<MenuItem>(
+    state.menuItem ?? MenuItem.SendReceive,
+  );
+
+  const isActive = (item: MenuItem) => {
+    return item === selectedItem;
+  };
+
+  let menuConf = {
+    title: 'MassaWallet',
+    logo: <MassaWallet />,
+    fullMode: true,
+  };
+
+  let menuItems = [
+    {
+      label: 'Wallet',
+      icon: <FiHome />,
+      active: false,
+      footer: false,
+      onClickItem: () => navigate(routeFor(MenuItem.Home)),
+    },
+    {
+      label: 'Transactions',
+      icon: <FiList />,
+      active: isActive(MenuItem.Transactions),
+      footer: false,
+      onClickItem: () => navigate(routeFor(MenuItem.Transactions)),
+    },
+    {
+      label: 'Send/Receive',
+      icon: <FiArrowUpRight />,
+      active: isActive(MenuItem.SendReceive),
+      footer: false,
+      onClickItem: () => navigate(routeFor(MenuItem.SendReceive)),
+    },
+    {
+      label: 'Contacts',
+      icon: <FiUsers />,
+      active: isActive(MenuItem.Contacts),
+      footer: false,
+      onClickItem: () => navigate(routeFor(MenuItem.Contacts)),
+    },
+    {
+      label: 'Assets',
+      icon: <FiDisc />,
+      active: isActive(MenuItem.Assets),
+      footer: false,
+      onClickItem: () => navigate(routeFor(MenuItem.Assets)),
+    },
+    {
+      label: 'Settings',
+      icon: <FiSettings />,
+      active: isActive(MenuItem.Settings),
+      footer: true,
+      onClickItem: () => navigate(routeFor(MenuItem.Settings)),
+    },
+    {
+      label: 'Light theme',
+      icon: <FiSun />,
+      active: isActive(MenuItem.LightTheme),
+      footer: true,
+      onClickItem: () => setSelectedItem(MenuItem.LightTheme),
+    },
+  ];
+
+  const accountsItems = accounts.map((account) => ({
+    icon: <FiUser size={32} />,
+    item: account.nickname,
+    // onClick: () => setSelectedAccount(account),
```

Comment:
why is it commented out? it's rarely a good practice to add commented out code, at least add a comment to explain why you commented it.

Reply:
legacy deleted in changes